staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -207,6 +207,11 @@ func (config *DirectClientConfig) ClientConfig() (*restclient.Config, error) {
 		mergo.Merge(clientConfig, serverAuthPartialConfig, mergo.WithOverride)
 	}
 
+	// in case of no persister we use noop one
+	if clientConfig.AuthConfigPersister == nil {
+		clientConfig.AuthConfigPersister = NoopPersister()
```

Comment:
rather than mutating clientConfig here, would it make more sense to modify GetAuthProvider to pass a no-op persister if needed? That would fix all code paths that reached that point, not just DirectClientConfig#ClientConfig

Reply:
good idea!

I did modify GetAuthProvider to do the job.